pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -180,6 +181,24 @@ private void initInstanceDataDir(File instanceDataDir) {
     }
   }
 
+  @Override
+  public List<File> getConsumerDirPaths() {
+    String consumerDirPath = _instanceDataManagerConfig.getConsumerDir();
+
+    if (consumerDirPath != null) {
+      return Collections.singletonList(new File(consumerDirPath));
+    } else {
+      List<File> consumerDirs = new ArrayList<>();
+      String instanceDataDir = _instanceDataManagerConfig.getInstanceDataDir();
+      for (String tableNameWithType : getAllTables()) {
+        String dirPath = instanceDataDir + File.separator + tableNameWithType + File.separator
+            + RealtimeTableDataManager.CONSUMERS_DIR;
+        consumerDirs.add(new File(dirPath));
+      }
+      return consumerDirs;
```

Comment:
should we store returnb value (consumerDirs) up so that we don't calculate these every time? We can add logging to it if it is calculated just once.

Reply:
my concern with that is that the stored consumerDirs will need to be updated with table changes on the instance. Thus, relying on table data managers to fetch paths to avoid that.